Source/gravity/Gravity.cpp

```diff
@@ -1377,21 +1377,109 @@ Gravity::interpolate_monopole_grav(int level, RealVector& radial_grav, MultiFab&
     int n1d = radial_grav.size();
 
     const Geometry& geom = parent->Geom(level);
-    const Real* dx = geom.CellSize();
-    const Real dr        = dx[0] / static_cast<Real>(gravity::drdxfac);
+    const auto dx = geom.CellSizeArray();
+    const Real dr = dx[0] / static_cast<Real>(gravity::drdxfac);
+
+    const auto problo = geom.ProbLoArray();
+
+    GpuArray<Real, 3> center;
+    ca_get_center(center.begin());
 
 #ifdef _OPENMP
 #pragma omp parallel
 #endif
     for (MFIter mfi(grav_vector, TilingIfNotGPU()); mfi.isValid(); ++mfi)
     {
-       const Box& bx = mfi.growntilebox();
-       ca_put_radial_grav(AMREX_ARLIM_ANYD(bx.loVect()), AMREX_ARLIM_ANYD(bx.hiVect()),
-                          ZFILL(dx), dr,
-                          BL_TO_FORTRAN_ANYD(grav_vector[mfi]),
-                          radial_grav.dataPtr(),
-                          ZFILL(geom.ProbLo()),
-                          n1d, level);
+        const Box& bx = mfi.growntilebox();
+
+        auto grav = grav_vector.array(mfi);
+        Real* const radial_grav_ptr = radial_grav.dataPtr();
+
+        // Note that we are interpolating onto the entire range of grav,
+        // including the ghost cells. Taking the absolute value of r ensures
+        // that we will get the correct behavior even for the ghost zones with
+        // negative indices, which have a reflecting boundary condition.
```

Comment:
I don't see where the abs() is taken

Reply:
This was a slightly out of date comment that comes from when I converted the 1D/2D/3D implementation to the dimension-agnostic form earlier this month. We do indirectly ensure that `r` is positive by computing sqrt(loc(:)**2), but in 1D this was just an `abs()`. I've removed the sentence just to avoid the confusion.